### DISCUSSION TITLE
"calling run inside a local task will have the same effect as calling runLocally"

### DISCUSSION THREAD

**Discussion by staabm** [deployphp/deployer#2381] [General]
just stumbled over this "note" in the docs
https://deployer.org/docs/tasks.html#local-tasks

I find it really supprising that `run` - one of the most often used commands - can change meaning depending on whether we are running a local task or not.

IMO the code would be better understandable/readable when `run` would throw a exception in a `local` task and `runLocally` would be required to use, no matter whether a task is local or not.

is this just "syntactic sugar" or is this magic-behaviour required/usefull?
is this something you are willing to change with the new majoar 7.x which is still in a kind of beta-state?

**Comment by antonmedv** [comment]
I also find local task not useful. What about deleting such feature in v7?

**Reply by staabm** [reply]
I am fine with deleting local() tasks. Stumbled over them today but what I actually need is once() which is also supported.


From my POV local() can go

**Reply by staabm** [reply]
(Or at least deprecated for now)

**Reply by staabm** [reply]
Do you agree with deprecating it first? Or should it be removed without prior deprecation?

**Reply by antonmedv** [reply]
I’m still thinking about it. There is a case there you want a task without host. For notify for example.

**Reply by staabm** [reply]
In which case one could still use runLocally() without marking the task as local(), no?

**Reply by antonmedv** [reply]
This is true.
